cmd/utils/flags.go

```diff
@@ -2076,7 +2076,7 @@ func tryMakeReadOnlyDatabase(ctx *cli.Context, stack *node.Node) ethdb.Database
 	// If datadir doesn't exist we need to open db in write-mode
 	// so database engine can create files.
 	readonly := true
-	if !common.FileExist(stack.ResolvePath("chaindata")) {
+	if !common.FileExist(stack.ResolvePath("chaindata/CURRENT")) {
```

Comment:
Perhaps we should use https://github.com/ethereum/go-ethereum/blob/master/core/rawdb/database.go

`if rawdb.HasPreexistingDb(stack.ResolvePath("chaindata")) != "" {`